apps/web/src/stores/documentVersions.ts

```diff
@@ -254,68 +263,111 @@ export default function useDocumentVersions(
     useLatitudeAction(assignDatasetAction, {
-      onSuccess: ({ data: document }) => {
-        if (!document) return
-
-        const prevDocuments = data || []
-        mutate(
-          prevDocuments.map((d) =>
-            d.documentUuid === document.documentUuid ? document : d,
-          ),
-        )
-      },
+      onSuccess: useCallback(
+        ({
+          data: document,
+        }: {
+          data: inferServerActionReturnData<typeof assignDatasetAction>
+        }) => {
+          if (!document) return
+
+          const prevDocuments = data || []
+          mutate(
+            prevDocuments.map((d) =>
+              d.documentUuid === document.documentUuid ? document : d,
+            ),
+          )
+        },
+        [data, mutate],
+      ),
     })
 
   const { execute: saveLinkedDataset, isPending: isLinkingDataset } =
     useLatitudeAction(saveLinkedDatasetAction, {
-      onSuccess: ({ data: document }) => {
-        if (!document) return
-
-        const prevDocuments = data || []
-        mutate(
-          prevDocuments.map((d) =>
-            d.documentUuid === document.documentUuid ? document : d,
-          ),
-        )
-      },
+      onSuccess: useCallback(
+        ({
+          data: document,
+        }: {
+          data: inferServerActionReturnData<typeof saveLinkedDatasetAction>
+        }) => {
+          if (!document) return
+
+          const prevDocuments = data || []
+          mutate(
+            prevDocuments.map((d) =>
+              d.documentUuid === document.documentUuid ? document : d,
+            ),
+          )
+        },
+        [data, mutate],
+      ),
     })
 
-  return {
-    data,
-    isValidating: isValidating,
-    isLoading: isLoading,
-    error: swrError,
-    createFile,
-    uploadFile,
-    renamePaths,
-    destroyFile,
-    destroyFolder,
-    updateContent,
-    isUpdatingContent,
-    assignDataset,
-    saveLinkedDataset,
-    mutate,
-    isAssigning: isAssigningDataset || isLinkingDataset,
-    isDestroying: isDestroyingFile || isDestroyingFolder,
-  }
+  return useMemo(
+    () => ({
+      data,
+      isValidating: isValidating,
+      isLoading: isLoading,
+      error: swrError,
+      createFile,
+      uploadFile,
+      renamePaths,
+      destroyFile,
+      destroyFolder,
+      updateContent,
+      isUpdatingContent,
+      assignDataset,
+      saveLinkedDataset,
+      mutate,
+      isAssigning: isAssigningDataset || isLinkingDataset,
+      isDestroying: isDestroyingFile || isDestroyingFolder,
+    }),
+    [
+      data,
+      isValidating,
+      isLoading,
+      swrError,
+      createFile,
+      uploadFile,
+      renamePaths,
+      destroyFile,
+      destroyFolder,
+      updateContent,
+      isUpdatingContent,
+      assignDataset,
+      saveLinkedDataset,
+      mutate,
+      isAssigningDataset,
+      isLinkingDataset,
+      isDestroyingFile,
+      isDestroyingFolder,
+    ],
+  )
```

Comment:
This store is used in many places. I reviewed it and tried to memoize all possible functions and the return so it's stable

Reply:
thanks!